### DISCUSSION TITLE
How to find the unique record in case of M-to-N relational query with include?

### DISCUSSION THREAD

**Discussion by palashCItobuz** [prisma/prisma#9457] [Q&A]
`Notifications` have `many-to-many` relation with `Users`. An event can be notified to multiple users. However, while returning the list of notifications for the user there is an array being returned for the relational `Users` data.

```prisma
model Notification {
  id                 Int             @id @default(autoincrement())
  vendorId      Int             @map("vendor_id")
  data             Json?
  userNotifications           UserNotification[]
}

model UserNotification {
  id              Int             @id @default(autoincrement())
  userId       Int             @map("user_id")
  user          User          @relation(fields: [userId], references: [id])
  eventId     Int              @map("event_id")
  event        Event         @relation(fields: [eventId], references: [id])
  createdAt DateTime   @default(now()) @map("created_at")
}

model User {
  id                                Int                   @id @default(autoincrement())
  admin                         Boolean          @default(false)
  email                          String              @unique
  name                          String
  userNotifications        UserNotification[]
}
```
However, it is so that a notification can only be ascribed to one user. So while showing the notification for the user, we don't need the array being returned when returned using `include : {  userNotifications: true }`. Tried using `take: 1` but it still returns an array.

```json
{
    "id": 1428,
    "userNotifications": [
       {
         "id": 1428,
         "userId": 30,
         "readAt": null,
         "flaggedAt": 1,
         "hideAt": null
       }
    ]
 },
{
  "id": 1425,
  "userNotifications": [
    {
      "id": 1425,
      "userId": 30,
      "readAt": "2021-09-23T05:45:10.405Z",
      "flaggedAt": 1,
      "hideAt": null
    }
  ]
},
.....
```
How do I return only 1 object instead of the array containing 1 object (it's always going to be 1) when I use 
```ts
let events = await context.prisma.notifications.findMany({
      include: {
        _count: true,
        userNotifications: {
          take: 1
        },
      },
      orderBy: [
        {
          createdAt: orderBy,
        },
      ],
      take,
      skip,
      where: whereClause,
    });

**Comment by ryands17** [comment]
@palashCItobuz 👋 

The reason you're getting an array is because you have a many side of the relation like this:

```prisma
model Notification {
  userNotifications           UserNotification[]
}
```

This will always return an array even if you're fetching a single record as it's a many relation.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
